### PR TITLE
raftstore/hibernate: wake up for more cases

### LINKED ISSUE #6668: Leader goes into Idle start after restarting
## Bug Report

Deployment: 1 tikv + 1 rngine
image
image

There is only one region in TiKV. After the TiKV instance is restarted, the peer in this TiKV campaigns to become a leader. Then it hibernated immediately and do not send any message to the learner. After 5min, the leader wakes up and finally send a heartbeat to learner.

### PR DESCRIPTION
**Pull request by BusyJay**
###  What have you changed?

This PR wakes up leader for more cases:
- Singleton can become leader in a tick, which breaks the hibernate check.
    This patch verifies if it's still follower to prevent it from hibernating.
- If leader sleeps during pre-transfer leader phase, it will keep sleep until other follower campaigns.
- If a learner disconnect for a while, it can't be prompted until next check invalid timeout.

Close #6668 

###  What is the type of the changes?
- Bugfix

###  How is the PR tested?
- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No.

###  Does this PR affect `tidb-ansible`?
No.

###  Refer to a related PR or issue link (optional)
#6668

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
@5kbpers @gengliqi PTAL

**Comment by BusyJay** [comment]
@5kbpers @gengliqi PTAL

File: components/raftstore/src/store/peer.rs [hunk 1]
```diff
@@ -734,6 +746,7 @@ impl Peer {
                 self.pending_messages.push(resp);
                 return Ok(());
             }
+            self.should_wake_up = state == LeaseState::Expired;
```
Comment (5kbpers) [inline components/raftstore/src/store/peer.rs]: I think leader should wake up once it receives `MsgReadIndex`, not only when lease has expired.

Comment (BusyJay) [reply components/raftstore/src/store/peer.rs]: If it's suspect, it's woken up already. See L2281.

**Pull request merged**